pkgs/development/interpreters/python/build-python-package-setuptools.nix

```diff
@@ -19,13 +21,16 @@ let
   # pip does the same thing: https://github.com/pypa/pip/pull/3265
   setuppy = ./run_setup.py;
 
+  setupPyDistFlagsString = lib.concatStringsSep " " setupPyDistFlags;
+  setupPyBuildExtString = lib.optionalString (setupPyBuildFlags != []) ("build_ext " + (lib.concatStringsSep " " setupPyBuildFlags));
+
 in attrs // {
   # we copy nix_run_setup over so it's executed relative to the root of the source
   # many project make that assumption
   buildPhase = attrs.buildPhase or ''
     runHook preBuild
     cp ${setuppy} nix_run_setup
-    ${python.pythonForBuild.interpreter} nix_run_setup ${lib.optionalString (setupPyBuildFlags != []) ("build_ext " + (lib.concatStringsSep " " setupPyBuildFlags))} bdist_wheel
+    ${python.pythonForBuild.interpreter} nix_run_setup ${setupPyDistFlagsString} ${setupPyBuildExtString} bdist_wheel
```

Comment:
If I read `nix_run_setup` docs correctly `setupPyDistFlags` here will be applied to _all_ commands, not just dist. In fact I need just that for Tensorflow but we need to either fix the option name to `setupPyCommonFlags` or force the arguments to be for `bdist_wheel` only by placing them after `bdist_wheel`.

Reply:
Could you give a link to the docs you've used?

Reply:
You are correct the name is a bit too specific.

Reply:
```
usage: nix_run_setup [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: nix_run_setup --help [cmd1 cmd2 ...]
   or: nix_run_setup --help-commands
   or: nix_run_setup cmd --help
```

From reading `run_setup.py` source code I assume this actually comes from setuptools.

Reply:
Yes, `setup.py` is distutils/setuptools and these options come from the `setup` function.